fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java

```diff
@@ -133,6 +134,18 @@ public Subspace getSecondarySubspace() {
     @Nonnull
     public abstract RecordCursor<IndexEntry> scanUniquenessViolations(@Nonnull TupleRange range, @Nullable byte[] continuation, @Nonnull ScanProperties scanProperties);
 
+    /**
+     * It scans the index and checks if the data (i.e. entries) is valid.
+     *
+     * It is not responsible for metadata validation, which is defined at
+     * {@link com.apple.foundationdb.record.metadata.IndexValidator}).
+     *
+     * @param continuation any continuation from a previous validation invocation
+     * @return a cursor over invalid index entries including reasons.
+     */
+    @Nonnull
+    public abstract RecordCursor<InvalidIndexEntry> validateEntries(byte[] continuation);
```

Comment:
My concerned was other kinds of validations in the future may not be able to take `ScanProperties`, for say, it may only be validated by scanning in a specific order. I don't have much knowledge about other types of indexes and the validations they need, so I tried to leave more flexibility to the validators.

Reply:
With great power comes great responsibility :-)  The direction is a valid concern, but I still feel like we may want to put some control back into the caller's hands as to how far the scan goes.  Maybe it makes sense to have to signatures, one with the `ScanProperties` and one with the default set of properties that you have here.  You can always throw a comment on the method that takes a `ScanProperties` that individual index implementations may not be able to adhere to all properties that are requested, such as `isReverse()`. 

Other than this, I think the rest of the PR looks good.

Reply:
Done.